File: frontend-app/Frontend/src/app/services/reviewService.ts
import client from '../api/client';
import { Review, ReviewRequest } from '../types/review';

export const reviewService = {
    getReviewsByListing: async (listingId: number): Promise<Review[]> => {
        const response = await client.get<Review[]>(`/api/reviews/listing/${listingId}`);
        return response.data;
    },

    createReview: async (review: ReviewRequest): Promise<Review> => {
        const response = await client.post<Review>('/api/reviews', review);
        return response.data;
    }
};

File: review-service/ReviewService/src/main/java/com/wedding/reviewService/service/ReviewService.java
package com.wedding.reviewService.service;

import com.wedding.reviewService.dto.ReviewRequest;
import com.wedding.reviewService.entity.Review;

import java.util.List;

public interface ReviewService {
    Review createReview(ReviewRequest reviewRequest, Long userId);

    List<Review> getReviewsByListing(Long listingId);
}
